### PR TITLE
oneOf tests for success-caused failures

### PR DESCRIPTION
**Pull request by RitoG09**
Added test cases for `oneOf` for now focusing on edge cases where validation fails due to multiple successful matches.
Related to issue: #120 
Planning to add similar tests for not and contains after review.

### CODE REVIEW

File: src/test-suite/tests/not.json [hunk 1]
```diff
@@ -24,6 +24,95 @@
...
+                "messageId": "required-success",
+                "messageParams": { "property": "a" },
+                "instanceLocation": "#",
+                "schemaLocations": ["#/not/required"]
+              }
+            ]
+          ]
+        }
+      ]
+    },
```
Comment (jdesrosiers) [inline src/test-suite/tests/not.json]: I see no difference between this test and the one before it. The `not` schemas are different, but there's nothing fundamentally different about what it's testing.
